### DISCUSSION TITLE
Using LPS Node firmware with ESP32 module combined with DWM1000 for TDoA

### DISCUSSION THREAD

**Discussion by anthony211212** [bitcraze/discussions#1072] [Q&A]
Hello everyone. Has anyone tried this code with the dwm1000 module in esp32? I'm thinking of using it for TDoA but I couldn't find a complete result. Does anyone have any suggestions? There are files such as lpsTdoaTag.c, uwb_tdoa_anchor2.c and uwb_tdoa_anchor3.c in the files, but I couldn't understand exactly what they are. I would appreciate if anyone can guide me. My goal is to use multiple tags and anchors with TDoA method. I am thinking of using esp32 for this. Thanks for your help in advance.

**Comment by knmcguire** [comment]
Hi. This is outside of the scope of this forum. We can only support fimware questions about our hardware, which is the [official LPS node](https://www.bitcraze.io/products/loco-positioning-node/). You are allowed to port the firmware to any hardware you want, but then we do expect you to be able to look into the code yourself to try to figure it out. From Bitcraze you can only expect very minimum support.

I'll refer to the answer we gave someone that is building their own platform based on the schematics of the crazyflie for a more complete answer for this reasoning: https://github.com/orgs/bitcraze/discussions/1037. Also I'd like to refer to our discussion rules: https://github.com/orgs/bitcraze/discussions/55

**Reply by anthony211212** [reply]
Thank you for the information.
